presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoDatabaseMetaData.java

```diff
@@ -1499,4 +1514,14 @@ private static void quoteStringLiteral(StringBuilder out, String value)
         }
         out.append('\'');
     }
+
+    private boolean stringIsLikePattern(String value) throws SQLException
+    {
+        if (value == null) {
+            return false;
+        }
+        Pattern p = Pattern.compile("[%_]");
+        Matcher m = p.matcher(value.replaceAll(getSearchStringEscape() + "\\.", ""));
```

Comment:
Use `Pattern.quote` to quote search string escape. Are you sure the "." was to be escaped?